### PR TITLE
Fix warp size

### PR DESCRIPTION
**Pull request by amd-sriram**
Replace C10_WARP_SIZE, constant WARP_SIZE, constant THREADS_PER_WARP, warp.*32 regex uses with at::cuda::warp_size() when not used __device__ or __global__.

Tested with docker 
`registry-sc-harbor.amd.com/framework/compute-rocm-dkms-no-npi-hipclang:16420_ubuntu22.04_py3.10_pytorch_lw_rocm7.0_internal_testing_2d567672`

Affected extensions:
1. fused rope - csrc/megatron/fused_rotary_positional_embedding.h
2. scaled_masked_softmax_cuda - , csrc/megatron/scaled_masked_softmax.h
3. generic_scaled_masked_softmax_cuda - csrc/megatron/generic_scaled_masked_softmax.h
4. scaled_upper_triang_masked_softmax_cuda - csrc/megatron/scaled_upper_triang_masked_softmax.h
5. group batch norm - apex/contrib/csrc/groupbn/batch_norm.h, apex/contrib/csrc/groupbn/batch_norm_add_relu.h
6. multihead attention - apex/contrib/csrc/multihead_attn/softmax.cuh
7. transducer - apex/contrib/csrc/transducer/transducer_joint_kernel.cu
8. xentropy - apex/contrib/csrc/xentropy/xentropy_kernel.cu
9. sync batch norm - csrc/welford.cu

The following UTs pass:

1. tests/L0/run_transformer/test_fused_rope.py
2. tests/L0/run_transformer/test_fused_softmax.py
3. apex/contrib/test/groupbn/test_groupbn.py
4. apex/contrib/test/groupbn/test_groupbn_channel_last.py
5. apex/contrib/test/multihead_attn/test_mha_fused_softmax.py
6. apex/contrib/test/transducer/test_transducer_joint.py
7. apex/contrib/test/transducer/test_transducer_loss.py
8. apex/contrib/test/test_label_smoothing.py
9. tests/distributed/synced_batchnorm/python_single_gpu_unit_test.py
10. tests/distributed/synced_batchnorm/single_gpu_unit_test.py
11. tests/distributed/synced_batchnorm/test_batchnorm1d.py

### CODE REVIEW

**Comment by amd-sriram** [comment]
! cherry-pick --onto release/1.4.0 release/1.5.0 release/1.6.0 release/1.7.0

**Comment by okakarpa** [comment]
Created branch [autogenerated/release/1.4.0_cherry-pick_pr-256](https://github.com/ROCm/apex/tree/autogenerated/release/1.4.0_cherry-pick_pr-256) and https://github.com/ROCm/apex/pull/257. It contains a merge conflict. Please resolve it

Created branch [autogenerated/release/1.5.0_cherry-pick_pr-256](https://github.com/ROCm/apex/tree/autogenerated/release/1.5.0_cherry-pick_pr-256) and https://github.com/ROCm/apex/pull/258

Created branch [autogenerated/release/1.6.0_cherry-pick_pr-256](https://github.com/ROCm/apex/tree/autogenerated/release/1.6.0_cherry-pick_pr-256) and https://github.com/ROCm/apex/pull/259

Created branch [autogenerated/release/1.7.0_cherry-pick_pr-256](https://github.com/ROCm/apex/tree/autogenerated/release/1.7.0_cherry-pick_pr-256) and https://github.com/ROCm/apex/pull/260
